moshi/moshi/server.py

```diff
@@ -223,27 +236,53 @@ def main():
     app = web.Application()
     app.router.add_get("/api/chat", state.handle_chat)
     static_path: None | str = None
+
     if args.static is None:
         log("info", "retrieving the static content")
         dist_tgz = hf_hub_download("kyutai/moshi-artifacts", "dist.tgz")
         dist_tgz = Path(dist_tgz)
         dist = dist_tgz.parent / "dist"
         if not dist.exists():
             with tarfile.open(dist_tgz, "r:gz") as tar:
-                tar.extractall(path=dist_tgz.parent)
+                safe_extract(tar, path=dist_tgz.parent)
         static_path = str(dist)
-    elif args.static != "none":
+    elif args.static.lower() != "none":
         # When set to the "none" string, we don't serve any static content.
         static_path = args.static
+
     if static_path is not None:
-        async def handle_root(_):
-            return web.FileResponse(os.path.join(static_path, "index.html"))
+        # Resolve the static path to an absolute path
+        static_path = Path(static_path).resolve()
 
-        log("info", f"serving static content from {static_path}")
+        # Ensure that the static path exists and is a directory
+        if not static_path.is_dir():
+            log("error", f"The provided static path is not a valid directory: {static_path}")
+            sys.exit(1)
+
+        # Optional: Define a base directory to restrict static content
+        # Uncomment and set to your base directory if needed
+        # base_static_dir = Path("/var/www/static").resolve()
+        # if not static_path.is_relative_to(base_static_dir):
+        #     log("error", "Attempt to access a directory outside the allowed base directory.")
+        #     sys.exit(1)
+
+        async def handle_root(_):
```

Comment:
As we do not pretend to be resilient to a wrongly formed `static`, I think I would rather have a mucher simpler implementation, closer to the original, simply adding before the serving:
```
assert (Path(static_path) / 'index.html').is_file(), "index.html missing in static folder."
```

Reply:
[![precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml/badge.svg)](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)

Reply:
[![precommit](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml/badge.svg)](https://github.com/kyutai-labs/moshi/actions/workflows/precommit.yml)